### PR TITLE
[Wip] Rio Transport

### PR DESCRIPTION
**Pull request by benaadams**
Sample `TransportChoiceApp` runs and works for benchmarking, otherwise is currently rougher than sand paper

`dotnet run -c Release -f netcoreapp2.0`

Then option 1 or 2

Not sure why it seems to listen on a different IP based on choice though (127.0.0.1 vs 127.0.0.127)

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel.Transport.Windows.Rio/Internal/Networking/RioCompletionThread.cs [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+            if (!_threadTcs.Task.IsCompleted)
+            {
+                DisposeConnections();
+            }
+        }
+
+        private void DisposeConnections()
+        {
+            //try
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Transport.Windows.Rio/Internal/Networking/RioCompletionThread.cs]: What is this dead code 😄 . @halter73 is working on moving shutdown to kestrel core so you won't have to handle this.

Comment (benaadams) [inline src/Microsoft.AspNetCore.Server.Kestrel.Transport.Windows.Rio/Internal/Networking/RioCompletionThread.cs]: Is like a todo area :)

File: src/Microsoft.AspNetCore.Server.Kestrel.Transport.Windows.Rio/RioTransport.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    .ConfigureAwait(false);
+            }
+            catch (AggregateException aggEx)
+            {
+                // An uncaught exception was likely thrown from the libuv event loop.
+                // The original error that crashed one loop may have caused secondary errors in others.
+                // Make sure that the stack trace of the original error is logged.
+                foreach (var ex in aggEx.InnerExceptions)
+                {
+                    Log.LogCritical("Failed to gracefully close Kestrel.", ex);
```
Comment (campersau) [inline src/Microsoft.AspNetCore.Server.Kestrel.Transport.Windows.Rio/RioTransport.cs]: `Kestrel`?

**Comment by muratg** [comment]
Closing this as "stale" :)  Thanks Ben!

**Pull request closed without merging**